### PR TITLE
systemvmtemplate: fix build and upgrade to debian 7.11 iso

### PR DESCRIPTION
**Pull request by rhtyd**
- Bumps base debian iso to version 7.11
- Upgrade ruby version to 2.3.0 (latest/stable)
- Fix Gemfile
- Update README

I've put pre-built `vhd-util` and `libvhd` used for converting xen template here: https://packages.shapeblue.com/systemvmtemplate/

cc @jburwell @agneya2001 @swill @wido @DaanHoogland and others -- please review

### CODE REVIEW

**Comment by rhtyd** [comment]
@swill this PR is ready for CI test run and merge, thanks

**Comment by swill** [comment]
how do I test this PR?

**Comment by rhtyd** [comment]
@swill you can use this PR on a branch and build systemvm templates. I've updated the README on how you can setup an environment to build yourself systemvmtemplates as well.

Here are two templates from two different PRs I built with this patch applied: https://packages.shapeblue.com/systemvmtemplate/custom/

File: tools/appliance/build.sh [hunk 1]
```diff
@@ -529,8 +534,6 @@ function vmware_export() {
     stage_vmx ${appliance_build_name}-vmware ${appliance_build_name}-vmware.vmdk
     ovftool ${appliance_build_name}-vmware.vmx ${appliance_build_name}-vmware.ova
   fi
-  bzip2 "${appliance_build_name}-vmware.vmdk"
-  mv "${appliance_build_name}-vmware.vmdk.bz2" dist/
```
Comment (rhtyd) [inline tools/appliance/build.sh]: Notes: we're removing this as we don't need vmdks for VMware, but OVAs

File: tools/appliance/definitions/systemvmtemplate/configure_systemvm_services.sh [hunk 2]
```diff
@@ -19,7 +19,7 @@
 set -e
 set -x
 
-CLOUDSTACK_RELEASE=4.6.0
+CLOUDSTACK_RELEASE=4.9.0
```
Comment (rhtyd) [inline tools/appliance/definitions/systemvmtemplate/configure_systemvm_services.sh]: Bumps version to match current version ^^

File: tools/appliance/definitions/systemvmtemplate/definition.rb [hunk 3]
```diff
@@ -27,15 +27,15 @@
 architectures = {
     :i386 => {
         :os_type_id => 'Debian',
-        :iso_file => 'debian-7.9.0-i386-netinst.iso',
-        :iso_src => 'http://cdimage.debian.org/cdimage/archive/7.9.0/i386/iso-cd/debian-7.9.0-i386-netinst.iso',
-        :iso_md5 => 'e101a11ddb31f85acef542df1a49bf57',
+        :iso_file => 'debian-7.10.0-i386-netinst.iso',
+        :iso_src => 'http://cdimage.debian.org/cdimage/archive/7.10.0/i386/iso-cd/debian-7.10.0-i386-netinst.iso',
+        :iso_md5 => '7fa97966e31afc705b3e115906cbcaa5',
```
Comment (rhtyd) [inline tools/appliance/definitions/systemvmtemplate/definition.rb]: Uses the latest/last Debian 7.x iso releases

**Comment by rhtyd** [comment]
Here is a build log to prove that build succeeds using a Ubuntu 16.04 slave box with VirtualBox 5.0:

[systemvmtemplate-build-consoleText.txt](https://github.com/apache/cloudstack/files/252729/systemvmtemplate-build-consoleText.txt)

On existing environment, not using the latest Ruby version; the only change they will need to do is update ruby to 2.3.0 (using rvm or global).

Build steps are simply:
```
cd tools/appliance
bundle install
bash -l build.sh systemvm64template
```

**Comment by rhtyd** [comment]
Ping @wido @swill @DaanHoogland @abhinandanprateek  can we get some review/lgtm thanks. This fixes build environment issues, updates readme and updates to use the last updated debian 7.x iso.

**Comment by DaanHoogland** [comment]
sure tag:easypr LGTM

**Comment by abhinandanprateek** [comment]
LGTM based on code review and template build.

File: tools/appliance/definitions/systemvmtemplate/definition.rb (hunk 3 above)
Comment (remibergsma) [inline tools/appliance/definitions/systemvmtemplate/definition.rb]: You can bump it to 7.11 now ;-)

File: tools/appliance/definitions/systemvmtemplate/configure_systemvm_services.sh (hunk 2 above)
Comment (remibergsma) [inline tools/appliance/definitions/systemvmtemplate/configure_systemvm_services.sh]: Browsed merged PRs but nothing is merged that requires this. Or do I miss something?

File: tools/appliance/definitions/systemvmtemplate/definition.rb (hunk 3 above)
Comment (remibergsma) [inline tools/appliance/definitions/systemvmtemplate/definition.rb]: Is i386 still being used?

**Comment by remibergsma** [comment]
@rhtyd Maybe we can replace all this with a Packer build? I already did that for Cosmic and I think it's at least 90% the same so it should be fairly easy. As a matter of fact, we use Cosmic systemvm templates with older CloudStack versions as well and that works fine. Along that way, a separate project with the systemvm would be best, as you could then iterate separately from CloudStack itself. A new version would just require a certain version of the systemvm template.
https://github.com/MissionCriticalCloud/systemvm-packer

**Comment by rhtyd** [comment]
@remibergsma sounds good to me, would you like to send a PR; port your work to master.

**Comment by DaanHoogland** [comment]
@remibergsma @rhtyd :+1: let's add the packer project and phase out the old template builds. more decoupling and (i suspect) simpler build process.

**Comment by wido** [comment]
Isn't this one ready to merge?

The Packer stuff from @remibergsma looks good, that makes building the SSVM templates a lot easier. But let's do that after the Debian bump?

**Comment by remibergsma** [comment]
@wido Let's bump Debian to 7.11 and let's not bump the systemvm template version, as nothing changed that requires users to replace all routers.

File: tools/appliance/definitions/systemvmtemplate/configure_systemvm_services.sh (hunk 2 above)
Comment (rhtyd) [inline tools/appliance/definitions/systemvmtemplate/configure_systemvm_services.sh]: This is not required for 4.9, but general improvements and updates.

File: tools/appliance/definitions/systemvmtemplate/definition.rb (hunk 3 above)
Comment (rhtyd) [inline tools/appliance/definitions/systemvmtemplate/definition.rb]: No, i386 is not used. Will bump it to 7.11 (7-11 ha :))

**Comment by rhtyd** [comment]
Alright, I've fixed the debian version without changing systemvmtemplate version.

@remibergsma we can get this merged for the time being, and let's have your packer stuff - that looks much better than what we've with ruby/veewee based.

**Comment by wido** [comment]
LGTM

The packer stuff can be done afterwards

**Comment by remibergsma** [comment]
Yeah, LGTM now with the changes. I didn't test a build, as I've no setup to do that.

**Comment by bvbharatk** [comment]
### ACS CI BVT Run
 **Sumarry:**
 Build Number 179
 Hypervisor xenserver
 NetworkType Advanced
 Passed=73
 Failed=0
 Skipped=3

_Link to logs Folder (search by build_no):_ https://www.dropbox.com/sh/yj3wnzbceo9uef2/AAB6u-Iap-xztdm6jHX9SjPja?dl=0


**Failed tests:**

**Skipped tests:**
test_vm_nic_adapter_vmxnet3
test_static_role_account_acls
test_deploy_vgpu_enabled_vm

**Passed test suits:**
test_deploy_vm_with_userdata.py
test_affinity_groups_projects.py
test_portable_publicip.py
test_vpc_vpn.py
test_over_provisioning.py
test_global_settings.py
test_scale_vm.py
test_service_offerings.py
test_routers_iptables_default_policy.py
test_routers.py
test_reset_vm_on_reboot.py
test_snapshots.py
test_deploy_vms_with_varied_deploymentplanners.py
test_login.py
test_list_ids_parameter.py
test_public_ip_range.py
test_multipleips_per_nic.py
test_regions.py
test_affinity_groups.py
test_network_acl.py
test_pvlan.py
test_volumes.py
test_nic.py
test_deploy_vm_root_resize.py
test_resource_detail.py
test_secondary_storage.py
test_vm_life_cycle.py
test_disk_offerings.py

**Comment by rhtyd** [comment]
@karuturi @jburwell this PR has enough review, can we merge this?

The changes are necessary without them the systemvm template appliance building fails, tested and built these templates: http://packages.shapeblue.com/systemvmtemplate/custom/biggervarlog/

**Pull request closed without merging**